src/replication.c

```diff
@@ -3415,7 +3447,7 @@ void syncWithPrimary(connection *conn) {
         if (err) goto write_error;
 
         server.repl_state = REPL_STATE_RECEIVE_AUTH_REPLY;
-        return;
+        goto cleanup;
```

Comment:
Do you want to move these changes to the PR that refactors `syncWithPrimary` too?

Reply:
yes, I will revert it